dvc/repo/gc.py

```diff
@@ -25,7 +34,19 @@ def gc(
     force=False,
     jobs=None,
     repos=None,
+    workspace=False,
 ):
+
+    # require `workspace` to be true to come into effect.
+    # assume `workspace` to be enabled if any of `all_tags`, `all_commits`,
+    # or `all_branches` are enabled.
+    _raise_error_if_all_disabled(
```

Comment:
I think, we should make `-w` mandatory (as it'll be symmetrical with `-c`).

Reply:
Looking into it again, I feel it's dangerous to do just `-c` to also delete from the current workspace. I'll propose changing it, so that `-c` will only delete from the remote and `-w` only from the local cache. And, combined both, they can have the previous behavior.

Thoughts?

Reply:
I would say we keep the current behavior in this PR. How `-c` should behave should be discussed separately.

Reply:
Kept the same behavior for now.